beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/service/PagingCleanupServiceTest.java

```diff
@@ -16,210 +16,83 @@
 package com.expediagroup.beekeeper.cleanup.service;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.doThrow;
-import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
 
-import java.time.Duration;
 import java.time.Instant;
 import java.time.LocalDateTime;
 import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.ArgumentCaptor;
-import org.mockito.Captor;
+import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.mock.mockito.MockBean;
-import org.springframework.test.annotation.DirtiesContext;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.TestPropertySource;
-import org.springframework.test.context.junit.jupiter.SpringExtension;
-import org.springframework.test.context.support.AnnotationConfigContextLoader;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
 
-import com.expediagroup.beekeeper.cleanup.TestApplication;
-import com.expediagroup.beekeeper.cleanup.path.PathCleaner;
+import com.expediagroup.beekeeper.cleanup.handler.GenericHandler;
+import com.expediagroup.beekeeper.core.error.BeekeeperException;
 import com.expediagroup.beekeeper.core.model.EntityHousekeepingPath;
-import com.expediagroup.beekeeper.core.model.PathStatus;
-import com.expediagroup.beekeeper.core.repository.HousekeepingPathRepository;
 
-@ExtendWith(SpringExtension.class)
 @ExtendWith(MockitoExtension.class)
-@TestPropertySource(properties = {
-    "hibernate.data-source.driver-class-name=org.h2.Driver",
-    "hibernate.dialect=org.hibernate.dialect.H2Dialect",
-    "hibernate.hbm2ddl.auto=create",
-    "spring.datasource.url=jdbc:h2:mem:beekeeper;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE;MODE=MySQL" })
-@ContextConfiguration(classes = { TestApplication.class },
-    loader = AnnotationConfigContextLoader.class)
-@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
-class PagingCleanupServiceTest {
+public class PagingCleanupServiceTest {
```

Comment:
A lot of the tests have been removed here. Why is this? 

I'm not entirely sure the paging will work with these changes. I remember we had some trouble at the beginning designing how the "find records" query works. What we settled with meant that we didn't have to call `pageable.next()`, we only had to do this if we are in dry run mode.

I think because this is now doing strictly unit tests, mocking what you expect the db to return, instead of what we had before (a sort of mini integration test against an in memory db), this change in functionality has been missed.

Reply:
IIRC my thought process was that since I had refactored `PagingCleanupService` to be more of an orchestrator of handlers, it didn't need to have that same scrutiny on the tests.

I do remember being slightly confused about the original query on how it progressed through in the default path w/o calling `next()`,so seeing that called out here means that I'd need to rethink this approach 😅.

Thinking about this a bit more for future proofing, if we're okay with converting the service into more of an orchestrator of handlers, we'd need to either consistently handle how we page through records at this level & refactor the call for `unreferenced` so it does need to call `next` or have it called out somewhere that any new lifecycle query has the expectation that it is written in a way that doesn't need to have `.next` called 

Completely open to suggestions on how to proceed on this part so it's not unintentionally breaking that original functionality.

Reply:
> I do remember being slightly confused about the original query on how it progressed through in the default path w/o calling next()

It progresses (if dryrun is off) because when a path is processed, its status is modified so "drops out" of the query result. Running the same query therefore will produce a different result each time without paging. The other case is where dryrun is enabled, and so no updates are made, so you have to do paging.

I imagine the `expired` handler will have to behave in a similar way? In which case, the distinction isn't about supporting the handlers, its about support dryruns. Does that make sense?

(I'm just thinking if there's possibly an edge case not covered by the original code: if an entire page of results are all originally `FAILED` and then _aren't_ updated because they fail again, I don't think we'll reach any more results...)

Reply:
ah okay. That makes way more sense now, and see how that could potentially be problematic. Let me go ahead and try and setup a test for that & revert the next call for dry run / prod runs

Reply:
Yes please, IIRC we actually had a bug that our original tests didn't catch as we made some incorrect assumptions about how the paging worked. We then fleshed them out quite a bit to reproduce the bug and prevent future regressions. So whatever it makes sense to keep please do.

Reply:
👍 done. Reverted these tests to be more "integration-like" so it'll actually use the in memory db setup.